bin/sozo/src/commands/options/transaction.rs

```diff
@@ -1,26 +1,48 @@
+use std::fmt::{Display, Formatter};
+
 use anyhow::{bail, Result};
-use clap::Args;
-use dojo_utils::{TxnAction, TxnConfig};
+use clap::builder::PossibleValue;
+use clap::{Args, ValueEnum};
+use dojo_utils::{EthFeeConfig, FeeConfig, StrkFeeConfig, TxnAction, TxnConfig};
 use starknet::core::types::Felt;
 
 #[derive(Debug, Args, Default)]
 #[command(next_help_heading = "Transaction options")]
 pub struct TransactionOptions {
+    #[arg(long)]
+    #[arg(help = "Fee token to use.")]
+    #[arg(default_value = "strk")]
+    pub fee_token: FeeToken,
+
     #[arg(long, value_name = "MULTIPLIER")]
-    #[arg(help = "The multiplier to use for the fee estimate.")]
+    #[arg(help = "The multiplier to use for the fee estimate (--fee-token eth).")]
     #[arg(long_help = "The multiplier to use for the fee estimate. This value will be used on \
                        the estimated fee which will be used as the max fee for the transaction. \
                        (max_fee = estimated_fee * multiplier)")]
     #[arg(conflicts_with = "max_fee_raw")]
+    #[arg(conflicts_with = "gas")]
+    #[arg(conflicts_with = "gas_price")]
     #[arg(global = true)]
     pub fee_estimate_multiplier: Option<f64>,
 
     #[arg(long)]
-    #[arg(help = "Maximum raw value to be used for fees, in Wei.")]
+    #[arg(help = "Maximum raw value to be used for fees, in Wei (--fee-token eth).")]
     #[arg(conflicts_with = "fee_estimate_multiplier")]
+    #[arg(conflicts_with = "gas")]
+    #[arg(conflicts_with = "gas_price")]
     #[arg(global = true)]
     pub max_fee_raw: Option<Felt>,
 
+    #[arg(long, help = "Maximum L1 gas amount (--fee-token strk).")]
+    #[arg(conflicts_with = "max_fee_raw")]
+    #[arg(conflicts_with = "fee_estimate_multiplier")]
+    pub gas: Option<u64>,
+
+    #[arg(long, help = "Maximum L1 gas price in STRK (--fee-token strk).")]
+    #[arg(conflicts_with = "max_fee_raw")]
+    #[arg(conflicts_with = "fee_estimate_multiplier")]
+    pub gas_price: Option<u128>,
+
```

Comment:
perhaps we should separate the options under the appropriate headings for better visual clarity

```console
Transaction options:
      --fee-token <FEE_TOKEN>  Fee token to use. [default: STRK] [possible values: ETH, STRK]
      --wait                   Wait until the transaction is accepted by the sequencer, returning the status and hash.
      --receipt                If --wait is set, returns the full transaction receipt. Otherwise, it is a no-op.
      --walnut                 Display the link to debug the transaction with Walnut.

Transaction options - ETH:
      --fee-estimate-multiplier <MULTIPLIER>
          The multiplier to use for the fee estimate (--fee-token eth).
      --max-fee-raw <MAX_FEE_RAW>
          Maximum raw value to be used for fees, in Wei (--fee-token eth).

Transaction options - STRK:
      --gas <GAS>              Maximum L1 gas amount (--fee-token strk).
      --gas-price <GAS_PRICE>  Maximum L1 gas price in STRK (--fee-token strk).
```

Reply:
Good suggestion, implemented in 1feac8e2fd7fdf01f98edc6428abf013b7363e97.